### PR TITLE
Introduce cache in provenance checks

### PR DESCRIPTION
**Pull request by fridex**
Depends-On: https://github.com/thoth-station/python/pull/12

### CODE REVIEW

**Comment by fridex** [comment]
We should signalize to user that results are cached in Thamos CLI.

**Comment by fridex** [comment]
recheck

File: thoth/user_api/api_v1.py [hunk 1]
```diff
@@ -127,9 +133,51 @@ def get_analyze_status(analysis_id: str):
...
+    try:
+        project = Project.from_strings(application_stack['requirements'], application_stack['requirements_lock'])
+    except ThothPythonException as exc:
+        _LOGGER.exception("Failed to parse project: %s", exc)
+        return {'parameters': parameters, 'error': f'Invalid application stack supplied: {exc}'}, 400
+    except Exception as exc:
+        _LOGGER.exception("Failed to parse project: %s", exc)
+        return {'parameters': parameters, 'error': 'Invalid application stack supplied'}, 400
+
+    graph = GraphDatabase()
```
Comment (goern) [inline thoth/user_api/api_v1.py]: can we move database queries to thoth.storages?

**Comment by fridex** [comment]
recheck

**Pull request merged**